x/protorev/keeper/epoch_hook.go

```diff
@@ -85,33 +85,38 @@ func (k Keeper) UpdatePools(ctx sdk.Context) error {
 }
 
 // GetHighestLiquidityPools returns the highest liquidity pools for all base denoms
+// by iterating through all pool modules, getting the total liquidity for each pool,
+// and updating the highest liquidity pools based upon comparing total liquidity.
 func (k Keeper) GetHighestLiquidityPools(ctx sdk.Context, baseDenomPools map[string]map[string]LiquidityPoolStruct) error {
-	// Get all pools
-	pools, err := k.gammKeeper.GetPoolsAndPoke(ctx)
-	if err != nil {
-		return err
-	}
-
-	// Iterate through all pools and find valid matches
-	for _, pool := range pools {
-		coins := pool.GetTotalPoolLiquidity(ctx)
-
-		// Pool must be active and the number of coins must be 2
-		if pool.IsActive(ctx) && len(coins) == 2 {
-			tokenA := coins[0]
-			tokenB := coins[1]
+	for _, poolModule := range k.poolmanagerKeeper.AllPoolModules() {
```

Comment:
@NotJeremyLiu on a second note we might not need to getAllPoolModules. We can only call GetPools and retrieve `GetTotalPoolliquidity`. The function doesnot exist right now but we are adding it in this PR https://github.com/osmosis-labs/osmosis/pull/4951/files#diff-aab00daa01569d8afdd695029e4cde3eba98d2c753e915d16558869ff4783858R671

Reply:
@stackman27 Oh yeah, this def solves our problem without us adding anything to poolManager in this PR. I'm cool with using that instead of making an AllPoolModules() function.